File: js/day1/main.js
const fs = require("fs");

function fuelFor(mass) {
    return Math.floor(mass / 3) - 2;
}

function recFuelFor(mass) {
    let total = 0;
    for (let fuel = fuelFor(mass); fuel > 0; fuel = fuelFor(fuel)) {
        total += fuel;
    }
    return total;
}

function main(argv) {
    const path = argv[0];
    const file = fs.readFileSync(path, "ascii");
    const lines = file.split("\n");
    lines.pop();

    const masses = lines.map((s) => parseInt(s, 10));
    const fuels = masses.map(fuelFor);
    console.log(fuels.reduce((acc, x) => acc + x));

    const recFuels = masses.map(recFuelFor);
    console.log(recFuels.reduce((acc, x) => acc + x));
}

main(process.argv.slice(2));

File: py/day1/main.py
def fuel(mass):
    return (mass // 3) - 2


def rec_fuel(mass):
    mass = fuel(mass)
    total = 0
    while mass > 0:
        total += mass
        mass = fuel(mass)
    return total


def main(args):
    total_fuel = 0
    total_rec_fuel = 0
    with open(args[1]) as f:
        for line in f:
            total_fuel += fuel(int(line))
            total_rec_fuel += rec_fuel(int(line))
    print(total_fuel)
    print(total_rec_fuel)


if __name__ == '__main__':
    import sys
    main(sys.argv)
